### ISSUE TITLE
[Bug]: GitHub Copilot Language Server loading issue.

### ISSUE THREAD

**Issue by pyby** [intitni/CopilotForXcode#477]
### Before Reporting

- [X] I have checked FAQ, and there is no solution to my issue
- [X] I have searched the existing issues, and there is no existing issue for my issue

### What happened?

I don't know when and how this issue appeared.

Not more Copilot proposition in Xcode.
In the application compagnion, I've an error in `Service`  and `Github Copilot`.

> The operation couldn't be completed. (LanguageServerProtocol.ServerError error 10.)

![Screenshot 2024-04-04 at 22 30 00](https://github.com/intitni/CopilotForXcode/assets/380522/919b5f4b-c005-41a1-923d-431556ec4fc5)

### How to reproduce the bug.

Tested on 2 different wifi environnements. Same issue.

Following https://github.com/intitni/CopilotForXcode/issues/463, I clean the cached folder `~/Library/Application Support/com.intii.CopilotForXcode/GitHub Copilot `. And reinstall the server. Same issue

### Relevant log output

```shell
error	22:38:41.658833+0200	Copilot for Xcode	failed to initialize: serverUnavailable
error	22:38:41.922504+0200	Copilot for Xcode	failed to initialize: serverUnavailable
error	22:38:42.109575+0200	Copilot for Xcode	failed to initialize: serverUnavailable
error	22:38:42.288285+0200	Copilot for Xcode	failed to initialize: serverUnavailable
error	22:38:42.500569+0200	Copilot for Xcode	failed to initialize: serverUnavailable
```

### macOS version

14.4

### Xcode version

15.3.0

### Copilot for Xcode version

0.31.3 (343)

**Comment by intitni** [comment]
It's a network issue. If you want to see the real cause, please turn the verbose log toggle on. You will see some logs that look like 

<img width="941" alt="Screenshot 2024-04-05 at 14 30 29" src="https://github.com/intitni/CopilotForXcode/assets/793147/fb7073de-a3f1-4546-bfe2-683dee073f1c">

If you don't see them, restart the app and try again.

**Comment by ErikMeinders** [comment]
I had the same issue and can confirm I had to install an up-to-date node version before I could continue. Didn't realise Mac OX Sonoma 14.4 came with an ancient version of node by default ...

**Comment by pyby** [comment]
Thanks @intitni , you got it. Node was broken on my Mac.

I don't from where the node installation was broken, anyway.

I updated node via `brew` and the command line works again:
```
brew install node
[…]
node -v
v21.7.2
```

Then I was able to run the server again and be authenticated.
![Screenshot 2024-04-05 at 10 46 09](https://github.com/intitni/CopilotForXcode/assets/380522/0bfa1d1a-690e-4755-bc9f-9c54fe62473b)

Verbose log mode for next time. 😌
